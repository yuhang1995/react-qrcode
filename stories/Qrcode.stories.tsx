import React from 'react';
import { Meta } from '@storybook/react';
import { Image, Canvas } from '../src';

const meta: Meta = {
    title: 'Welcome',
    component: Image,
    argTypes: {
        children: {
            control: {
                type: 'text',
            },
        },
    },
    parameters: {
        controls: { expanded: true },
    },
};

export default meta;

export const Base = () => {

    return (
        <div style={{ width: '500px', height: '500px', background: 'gray', display: 'flex', alignItems: 'center', justifyContent: 'center' }}>
            <Image
                text='https://github.com/yuhang1995/react-qrcode'
                options={{
                    type: 'image/jpeg',
                    margin: 3,
                    scale: 4,
                    width: 200,
                }}
            />
        </div>
    )
}

export const CanvasQrCode = () => {
    return (
        <div style={{ width: '500px', height: '500px', background: 'gray', display: 'flex', alignItems: 'center', justifyContent: 'center' }}>
            <Canvas
                text='https://github.com/yuhang1995/react-qrcode'
                options={{
                    margin: 3,
                    scale: 4,
                    width: 200,
                }}
            />
        </div>
    )
}
