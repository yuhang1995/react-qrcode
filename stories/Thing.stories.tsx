import React from 'react';
import { Meta } from '@storybook/react';
import { Image, Canvase } from '../src';

const meta: Meta = {
    title: 'Welcome',
    component: Image,
    argTypes: {
        children: {
            control: {
                type: 'text',
            },
        },
    },
    parameters: {
        controls: { expanded: true },
    },
};

export default meta;

export const Base = () => {
    // const ref = React.useRef<HTMLImageElement>(null)
    const ref = React.createRef<HTMLImageElement>()

    return (
        <div style={{ width: '500px', height: '500px', background: 'gray', display: 'flex', alignItems: 'center', justifyContent: 'center' }}>
            <Test />
            <Image
                ref={ref}
                text='https://github.com/bunlong/next-qrcode'
                options={{
                    type: 'image/jpeg',
                    // rendererOpts: {
                    //     quality: 0.3,
                    // },
                    margin: 3,
                    scale: 4,
                    width: 200,
                    // color: {
                    //     dark: '#010599FF',
                    //     light: '#FFBF60FF',
                    // },
                }}
            />
        </div>
    )
}
export const CanvaseQrCode = () => {

    return (
        <div style={{ width: '500px', height: '500px', background: 'gray', display: 'flex', alignItems: 'center', justifyContent: 'center' }}>
            <Canvase
                text='https://github.com/bunlong/next-qrcode'
                options={{
                    margin: 3,
                    scale: 4,
                    width: 200,

                }}
            />
        </div>
    )
}


const Test = React.forwardRef<HTMLButtonElement, {}>(function Test(_, ref) {

    return <button ref={ref}>+++</button>
})
