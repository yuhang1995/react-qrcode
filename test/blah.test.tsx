import React from 'react';
// import * as ReactDOM from 'react-dom/client';
import { Base } from '../stories/Qrcode.stories';
import renderer from 'react-test-renderer';


describe('Base', () => {
    it('renders without crashing', () => {
        //         const div = document.createElement('div');
        //         const root = ReactDOM.createRoot(div);
        //         root.render(<Base />);
        //         root.unmount();
        const component = renderer.create(
            <Base />,
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        // // manually trigger the callback
        // renderer.act(() => {
        //     tree.props.onMouseEnter();
        // });
        // // re-rendering
        // tree = component.toJSON();
        // expect(tree).toMatchSnapshot();

        // // manually trigger the callback
        // renderer.act(() => {
        //     tree.props.onMouseLeave();
        // });
        // re-rendering
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

