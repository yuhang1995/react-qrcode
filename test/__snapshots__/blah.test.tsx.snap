// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base renders without crashing 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "background": "gray",
      "display": "flex",
      "height": "500px",
      "justifyContent": "center",
      "width": "500px",
    }
  }
>
  <img
    src=""
  />
</div>
`;

exports[`Base renders without crashing 2`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "background": "gray",
      "display": "flex",
      "height": "500px",
      "justifyContent": "center",
      "width": "500px",
    }
  }
>
  <img
    src=""
  />
</div>
`;
